### PR TITLE
Added initial association timeout

### PR DESCRIPTION
**Pull request by mjanczykowski**

### CODE REVIEW

File: common/src/main/resources/supervisor.conf [hunk 1]
```diff
@@ -9,4 +9,4 @@ worker {
   nodes = 2
   areas_per_node = 2
 }
-supervisor.waiting-for-workers-timeout-seconds = 15
\ No newline at end of file
+supervisor.waiting-for-workers-timeout-seconds = 60
```
Comment (salceson) [inline common/src/main/resources/supervisor.conf]: We have some inconsistent ways of naming config values. Above we see `names-like-that`, `names_like_that`, below we see `names.like.that`. We need to decide one format.

File: common/src/main/scala/miss/worker/WorkerActor.scala [hunk 2]
```diff
@@ -9,14 +9,17 @@ import miss.worker.WorkerActor.{Data, State}
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
-class WorkerActor(supervisorPath: String, retryIntervalSeconds: Long) extends FSM[State, Data] {
+class WorkerActor(supervisorPath: String, retryIntervalSeconds: Long, supervisorAssociationTimeoutSeconds: Long) extends FSM[State, Data] {
```
Comment (salceson) [inline common/src/main/scala/miss/worker/WorkerActor.scala:12]: Long line (140 characters).

File: common/src/main/scala/miss/worker/WorkerActor.scala [hunk 3]
```diff
@@ -9,14 +9,17 @@ import miss.worker.WorkerActor.{Data, State}
...
 
   import WorkerActor._
   import context.dispatcher
 
   val supervisor = context.actorSelection(supervisorPath)
 
+  private val ASSOCIATION_TIMER = "association-timer"
```
Comment (salceson) [inline common/src/main/scala/miss/worker/WorkerActor.scala]: As far as I know, constants in Scala are named like this: `AssociationTimer`. This, of course is a convention. Are we using constants named like `ASSOCIATION_TIMER` everywhere?

File: common/src/main/scala/miss/worker/WorkerActor.scala [hunk 4]
```diff
@@ -69,9 +78,11 @@ object WorkerActor {
 
   case object Terminate extends SerializableMessage
 
+  case object AssociationTimedOut
+
   case object TerminateSystem
 
-  def props(supervisorPath: String, retryIntervalSeconds: Long): Props = Props(classOf[WorkerActor], supervisorPath, retryIntervalSeconds)
+  def props(supervisorPath: String, retryIntervalSeconds: Long,supervisorAssociationTimeoutSeconds: Long): Props = Props(classOf[WorkerActor], supervisorPath, retryIntervalSeconds, supervisorAssociationTimeoutSeconds)
```
Comment (salceson) [inline common/src/main/scala/miss/worker/WorkerActor.scala]: Very long line, unformatted.

**Review by salceson — CHANGES_REQUESTED** [review]

File: common/src/main/resources/supervisor.conf (hunk 1 above)
Comment (mjanczykowski) [reply common/src/main/resources/supervisor.conf]: I'll use underscore here, as I don't want to edit scripts.

File: common/src/main/scala/miss/worker/WorkerActor.scala (hunk 3 above)
Comment (mjanczykowski) [reply common/src/main/scala/miss/worker/WorkerActor.scala]: fixed here + in Supervisor

File: common/src/main/scala/miss/worker/WorkerActor.scala (hunk 4 above)
Comment (mjanczykowski) [reply common/src/main/scala/miss/worker/WorkerActor.scala]: fixed

**Review by salceson — APPROVED** [review]

**Pull request merged**